File: CSC330/poker/Java/Card.java
public class Card {
    private final String rank;
    private final String suit;

    public Card(String cardStr) {
        this.rank = cardStr.substring(0, 1); // First character
        this.suit = cardStr.substring(1); // Second character
    }

    public String getRank() {
        return rank;
    }

    public String getSuit() {
        return suit;
    }

    public int getRankValue() {
        return "23456789TJQKA".indexOf(rank);
    }

    public int getSuitValue() {
        return "SHCD".indexOf(suit);
    }

    public String toString() {
        return rank + suit;
    }
}

File: CSC330/poker/Fortran/card.f90
module DeckModule
    implicit none

    type :: Card
        character(len=2) :: rank
        character(len=1) :: suit
    contains
        procedure :: initializeCard
        procedure :: displayCard
    end type Card

contains

    ! Initialize a Card from a string representation
    subroutine initializeCard(this, cardStr)
        class(Card), intent(inout) :: this
        character(len=*), intent(in) :: cardStr

        this%rank = cardStr(1:2)
        this%suit = cardStr(3:3)
    end subroutine initializeCard

    ! Display the card
    subroutine displayCard(this)
        class(Card), intent(in) :: this
        print *, "Card: ", trim(this%rank), " of ", trim(this%suit)
    end subroutine displayCard

    type :: Deck
        type(Card), allocatable :: cards(:)
        integer :: size
    end type Deck

    ! Other subroutines like shuffle, dealCard, and readHandsFromFile

end module DeckModule

